### PR TITLE
Only display event locations with events

### PR DESCRIPTION
**Pull request by moritzcopter**
See this card: https://trello.com/c/94d31ca8/62-improve-local-group-events-filter

### CODE REVIEW

File: web/app/themes/xrnl/archive-meetup_events.php [hunk 1]
```diff
@@ -24,33 +33,41 @@
 	'orderby' => 'meta_value',
 	'meta_key' => 'event_start_date',
 	'order' => 'ASC',
+	's' => '-cancelled -afgelast',
```
[the remark this answers is not in the document]
Comment (moritzcopter) [reply web/app/themes/xrnl/archive-meetup_events.php:36]: see "Search parameters" here: https://developer.wordpress.org/reference/classes/wp_query/

[the remark this answers is not in the document]
Comment (moritzcopter) [reply web/app/themes/xrnl/archive-meetup_events.php:36]: > A much more efficient approach would be to execute a single SQL query that returns all locations with events and the number of events at each location.

I actually thought the same and played around a bit:

`   global $wpdb;
    $cities = $wpdb->get_results("
        select cities.meta_value as city, dates.meta_value as date
        from $wpdb->postmeta cities inner join $wpdb->postmeta dates on cities.post_id = dates.post_id
        where 
        cities.meta_key = 'venue_city' and cities.meta_value != ''
        and
        dates.meta_key = 'event_start_date' and dates.meta_value >= .$today.");
    
        return $cities;`

Unfortuntely the date comparison does not work for some reason so I couldnt get it to work..

[the remark this answers is not in the document]
Comment (SamLubbers) [reply web/app/themes/xrnl/archive-meetup_events.php:36]: > You are right - see "Search parameters" here: https://developer.wordpress.org/reference/classes/wp_query/

Thanks for the link! It was impossible for me to find it.

[the remark this answers is not in the document]
Comment (SamLubbers) [reply web/app/themes/xrnl/archive-meetup_events.php:36]: > Unfortuntely the date comparison does not work for some reason so I couldnt get it to work..

That's cool. We'll try to make it work if it ain't fast enough.

**Pull request merged**